### PR TITLE
Add casting for Finder.

### PR DESCRIPTION
**Pull request by dereuromark**
With stricter method signatures this becomes necessary for Finder usage.

Resolves 
- https://github.com/FriendsOfCake/search/issues/338
- https://github.com/FriendsOfCake/search/issues/341

We can release a new minor 7.1.0

### CODE REVIEW

File: src/Model/Filter/Finder.php [hunk 1]
```diff
@@ -37,6 +38,18 @@ public function process(): bool
         foreach ($map as $to => $from) {
             $args[$to] = $args[$from] ?? null;
         }
+        $casts = $this->getConfig('cast');
+        foreach ($casts as $field => $toType) {
+            $value = $args[$field] ?? null;
+
+            if (is_callable($toType)) {
```
Comment (ADmad) [inline src/Model/Filter/Finder.php:45]: The type setting and calling the callable should be skipped if the value is null, otherwise the null will get cast to integer 0 for e.g. There should also be a test for it.

Comment (dereuromark) [reply src/Model/Filter/Finder.php:45]: I just checked and added tests for both normal and callable using null input.
And it works as expected. How would we get an issue that you describe?

Comment (ADmad) [reply src/Model/Filter/Finder.php:45]: > I also added tests for when empty (request) strings need to be mapped to nullish value for ?type args.

I don't see that being useful in practice as `Processor::_extractParams()` filters out all falsey values by default.

The change I was suggesting was this:

```php
if ($value !== null) {
    if (is_callable($toType)) {
        $value = $toType($value);
    } else {
        settype($value, $toType);
    }
}

Skipping the casting all together if no value has been provided aka `null`.
```

Comment (ADmad) [reply src/Model/Filter/Finder.php:45]: The various tests cases you have setting which have `$filter->setArgs(['foo' => null]);` have negligible practical value and filtering records with a a column matching null / empty string is rarely needed. For those rare cases people can just use the `Callback` filter and even for that to work they will have to first change the empty values filtering.

**Comment by dereuromark** [comment]
Adjusted as per feedback

File: src/Model/Filter/Finder.php [hunk 2]
```diff
@@ -37,6 +38,21 @@ public function process(): bool
...
             $args[$to] = $args[$from] ?? null;
         }
+        $casts = $this->getConfig('cast');
+        foreach ($casts as $field => $toType) {
+            $value = $args[$field] ?? null;
+            if ($value === null) {
+                continue;
+            }
+
+            if (is_callable($toType)) {
```
Comment (ADmad) [inline src/Model/Filter/Finder.php:48]:
```suggestion
            if ($toType instanceof \Closure) {
```
I highly dislike `is_callable()` and doubt anyone will use anything besides a closure here.
